### ISSUE TITLE
error: [Errno 2] No such file or directory: 'ant': 'ant'  when building the python package

### ISSUE THREAD

**Issue by KOLANICH** [jpype-project/jpype#362]
```
Collecting JPype1@ git+https://github.com/jpype-project/jpype#egg=JPype1;platform_python_implementation=="CPython" from git+https://github.com/jpype-project/jpype#egg=JPype1;platform_python_implementation=="CPython"
Cloning https://github.com/jpype-project/jpype to /tmp/pip-install-nuvcpjfk/JPype1
Requirement already satisfied, skipping upgrade: setuptools in /usr/local/lib/python3.7/site-packages (40.5.0)
Building wheels for collected packages: JPype1
Running setup.py bdist_wheel for JPype1: started
Running setup.py bdist_wheel for JPype1: finished with status 'error'
Complete output from command /usr/local/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-nuvcpjfk/JPype1/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-eqdmoyft --python-tag cp37:
running bdist_wheel
running build
running build_py
creating build
creating build/lib.linux-x86_64-3.7
creating build/lib.linux-x86_64-3.7/jpype
copying jpype/_darwin.py -> build/lib.linux-x86_64-3.7/jpype
copying jpype/_jvmfinder.py -> build/lib.linux-x86_64-3.7/jpype
<.....>
running build_ext
running build_java
ant -Dbuild=../build/lib -f native/build.xml
error: [Errno 2] No such file or directory: 'ant': 'ant'
```

Is ant mandatory?

**Comment by Thrameos** [comment]
Ant is required to compile the java thunk code (java classes that are stored within the shared library) and to build the test bench.  For a long time the thunk code was just shipped as a precompiled java class embedded in the header so it was only used for testing.  This lead to the java code that we were depending upon to be lost complete.   My long term plan is to move some of the support classes from the module into java so that means we need to build more thunk code. 

There are two solutions we can use to remove the dependency at least for the thunk code.  
1) Compile the code manually using javac rather than ant.  The ant code we are using is pretty simple resulting in a few classes being built and potentially a jar.  But if we do that we would have to handle all the special cases about finding the java compiler.  Not hard but does add edge cases.
2) Ship the thunks like we did in the past.  But this means we would need both the ability to build the thunks and the ability to store the prebuild thunks when we cut a distribution.  Doable but again more work in maintenance. 

Given that we already require ant to test to code and given that at times it is necessary to see and debug stuff happening within the thunks I felt it would be much better if we build the thunks when we use it rather than having magic shipped in embedded header files.  It was also a requirement of my organization that code be entirely in source code for inspection but that is a local site issue. 

Thus at least currently ant is mandatory.

**Comment by KOLANICH** [comment]
Could you add this info into `install.rst`?

**Comment by anilpai** [comment]
I had a similar issue on Mac OS

I did following steps:

`brew install ant`

and I modified the MANIFEST.in (to include build.xml and java folder)

```
recursive-include native *.h
recursive-include examples *
recursive-include native *.xml
recursive-include native *
include *.rst doc/* LICENSE
# include everything under test/
graft test
graft setupext
global-exclude *.pyc
prune test/classes
```

**Comment by KOLANICH** [comment]
BTW, should we move these support classes into a separate package?

**Comment by Thrameos** [comment]
Which support classes?   The setupext package is just a bunch of classes where we extend setup for purposes of applying our customization.  The python setup does not have any support for mixed java compiling.  I am not sure any of it will be useful to others.  Though I guess if someone was building a package using jpype they may need to do similar things.  However, as it stands the setupext code is not really general use.  I didn't put any consideration into how to make it reusable and half the stuff is just hardcoding what is needed to get our stuff working.  At best setupext serves as examples of how to customize setup for a mixed java project.

**Comment by KOLANICH** [comment]
>Which support classes?

The ones written in java. I guess they are updated rarely (
>Latest commit 490eb65 on Jun 6

), so moving them into a separate package would mean that CI doesn't need `ant` in order to build jpype.

**Comment by Thrameos** [comment]
Fixed in #428
